Gems/Atom/Feature/Common/Assets/Materials/Special/ShadowCatcher.shader

```diff
@@ -11,7 +11,7 @@
 
     "RasterState": { "CullMode": "None" },
 
-    "BlendState": {
+    "GlobalTargetBlendState": {
```

Comment:
What happens when a .shader file uses the old "BlendState" key? What error does the AP report? If we are going to break old data files we should at least try to make sure there is a clear path to fixing the issue. Ideally, we should detect the situation, fail the job, and report an error telling the user exactly what to do about it: rename BlendState to GlobalTargetBlendState.

Reply:
That's a very good point, but I'm not sure how we can catch that. The deserializer will probably put a generic message saying it cannot unmarshal some of the data, but I don't know how we can catch that later in the process. I'll drop a line to sig-core to see if there is a way.

Reply:
It does not look like BlendState was removed, but GlobalTargetBlendState was added. If they both exist and work then nothing related to serialization needs to change. If something was being replaced or renamed that needed a conversion path you would have to create a custom JSON serializer.

Reply:
I'm not sure I understand what you meant. In that case `BlendState` changed type and what was `BlendState` became `GlobalTargetBlendState`.

Reply:
First, I would try temporarily reverting one of your .shader file changes and just see what happens. Check what kind of error message gets reported. If it's clear enough, we can send an impactful-changes message on Discord, and write wiki page about it with quotes of the specific error messages. Hopefully that's discoverable enough and the number of new .shader files out in the wild are small enough, that we can go with this approach.

If that's not good enough, then we can do one of the following...

1. As @gadams3 mentioned, we could make a custom JSON serializer for the ShaderSourceData class. It would be nice to avoid this if possible because there would be a lot of boilerplate code for such a narrow issue.
2. Or we could make a new custom struct that has fields from both BlendState and TargetBlendState struct so it could be used for either one, and ShaderAssetBuilder can figure out which fields to read from. It would be nice to avoid this if possible because, while it's much less code than option 1, it is rather clunky. (If it's not clear what I mean here I can provide more detail).

Reply:
It will spit a bunch or warnings when parsing the shader:
```
Skipped unrecognized field '/BlendState/Enable' from AssetBuilder
```
Although not "in your face' it might be enough for a user to track down the issue. That being said many of our shaders are throwing warnings that could hide it in the log.

I actually had something like 2 in place in the beginning, but I went against it as I believe it's too early in the engine lifecycle to introduce these kind of hacky constructs to shield user code. I went through the known partners repo to check for `BlendState` usage and couldn't find any, I would say let's submit an impactful notice and push it as is.

Reply:
I'm fine with that.

Reply:
Actually, there might be one thing we can do. By default the JSON serializer only reports warnings for skipped fields. We can force those to be errors for .shader files. For an example, see the function "LoadMaterialTypeSourceData", where it uses JsonReportingHelper. Notice the difference in "LoadShaderDataJson". We just need to add 
```
                if (reportingHelper.ErrorsReported())
                {
                    return AZ::Failure();
                }
```

Reply:
Correct me if I'm getting that wrong, but would that only activate that behaviour for that file? Wouldn't it be better to have it everywhere?

Reply:
Turns out unrecognized fields were not properly reported as warning, only logged as such. Changed that as well.